### PR TITLE
Fix D401 pydocstyle issues

### PR DESCRIPTION
**Pull request by TjuAachen**
As the previous [fix](https://github.com/apache/iceberg/pull/8181) has a merge conflict, this new pull request is created to fix D401 violations of [#7776](https://github.com/apache/iceberg/issues/7776). Hopefully, it can be reviewed and merged to the master branch soon.

### CODE REVIEW

File: python/tests/conftest.py [hunk 1]
```diff
@@ -1484,21 +1484,21 @@ def fixture_aws_credentials() -> Generator[None, None, None]:
...
-    """Mocked S3 client."""
+    """Yield a mocked S3 client."""
     with mock_s3():
         yield boto3.client("s3", region_name="us-east-1")
 
 
 @pytest.fixture(name="_glue")
 def fixture_glue(_aws_credentials: None) -> Generator[boto3.client, None, None]:
-    """Mocked glue client."""
+    """Yiled a mocked glue client."""
```
Comment (Fokko) [inline python/tests/conftest.py:1494]:
```suggestion
    """Yield a mocked glue client."""
```

**Review by Fokko — APPROVED** [review]
One typo, apart from that it looks great. Thanks @TjuAachen !

**Comment by Fokko** [comment]
Thanks @TjuAachen for fixing this! And @mhmohona for doing the initial work 🙏🏻

File: python/tests/conftest.py (hunk 1 above)
Comment (TjuAachen) [reply python/tests/conftest.py]: Good catch! Thanks for your review.

Comment (Fokko) [reply python/tests/conftest.py]: Any time, I should thank you for picking this up!

**Pull request merged**